### PR TITLE
fix: use deployment status for protx revoke/update_service BLS scheme

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Summary

Fix "bad-protx-sig" error when revoking or updating service for masternodes registered with legacy BLS keys on V19-active networks.

**Root Cause:** The RPC was changed to determine `ptx.nVersion` and signing scheme based on the masternode's state version. However, the validator uses the global BLS scheme flag for verification. This caused a mismatch for legacy BLS masternodes on V19-active networks:

| Step | v22.1.3 (works) | v23 (broken) |
|------|-----------------|--------------|
| `ptx.nVersion` | Based on V19 deployment (→ BasicBLS) | Based on MN state (→ LegacyBLS for old MNs) |
| Signing scheme | `!isV19active` (→ BasicBLS) | `ptx.nVersion == LegacyBLS` (→ LegacyBLS) |
| Verification | Global scheme (BasicBLS) | Global scheme (BasicBLS) |
| Result | Match ✓ | **Mismatch → FAIL** |

**Fix:** Restore v22.1.3 behavior - determine version and signing scheme based on network deployment status, not masternode state.

## Changes

- `src/rpc/evo.cpp`: 
  - `protx_revoke`: Use V19 deployment status for version and signing scheme
  - `protx_update_service`: Remove `is_basic_override` parameter to use deployment status

## Test plan

- [x] Added regression test `test/functional/rpc_protx_legacy_bls.py`
- [x] Test verifies `protx update_service` works for legacy BLS MNs after V19
- [x] Test verifies `protx revoke` works for legacy BLS MNs after V19
- [ ] Run full test suite

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: test/functional/rpc_protx_legacy_bls.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+The fix ensures that protx commands use the network deployment status
+(V19 active = basic scheme) rather than the MN's state version.
+'''
+
+from test_framework.test_framework import DashTestFramework, MasternodeInfo
+from test_framework.util import assert_equal
+
+
+class ProtxLegacyBLSTest(DashTestFramework):
```
Comment (knst) [inline test/functional/rpc_protx_legacy_bls.py:31]: feature_dip3_v19.py already have some limited tests for v19 activation and combination of legacy / basic RPCs.

It includes even tests for `protx revoke`:

        self.log.info(f"Trying to revoke proTx:{self.mninfo[-1].proTxHash}")
        self.test_revoke_protx(evo_info_3.nodeIdx, self.mninfo[-1])

Could feature_dip3_v19.py be improved by adding this particular corner case instead creating a brand new functional test just for one corner case?

File: src/rpc/evo.cpp [hunk 2]
```diff
@@ -1260,9 +1263,7 @@ static RPCHelpMan protx_revoke()
         throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("masternode %s not found", ptx.proTxHash.ToString()));
     }
 
-    ptx.nVersion = ProTxVersion::GetMaxFromDeployment<CProUpRevTx>(
-        WITH_LOCK(::cs_main, return chainman.ActiveChain().Tip()), chainman,
-        /*is_basic_override=*/dmn->pdmnState->nVersion >= ProTxVersion::BasicBLS);
+    ptx.nVersion = ProTxVersion::GetMax(isV19active, /*is_extended_addr=*/false);
```
Comment (UdjinM6) [inline src/rpc/evo.cpp:1266]: ~This breaks `revoke` for future `ExtAddr` masternodes.~ Actually, no, only `CProRegTx` and `CProUpServTx` are eligible for `ExtAddr` version.
